src/3_branch.js

```diff
@@ -587,6 +592,41 @@ Branch.prototype['link'] = wrap(callback_params.CALLBACK_ERR_DATA, function(done
 	});
 });
 
+/**
+ * @function Branch.getReferringLink
+ * @param {Object} data - _required_ - link data and metadata.
+ * @param {function(?Error,String=)} callback - _required_ - returns a string of the Branch deep linking URL.
+ *
+ * If the session was opened from a referring link, this method will return the referring link click, which gives you the ability to continue the click flow.
+ *
+ * #### Example
+ * ```js
+ * branch.getReferringLink(function(err, data) {
+ *     console.log(err, link);
+ * });
+ * ```
+ *
+ * ##### Callback Format
+ * ```js
+ * callback(
+ *     null,
+ *     'https://bnc.lt/c/3HZMytU-BW' // Branch referring link click URL
+ * );
+ * ```
+ *
+ * ___
+ *
+ * ## Sharing links via SMS
+ */
+Branch.prototype['getReferringLink'] = wrap(callback_params.CALLBACK_ERR_DATA, function(done) {
+	var click_url = utils.readKeyValue('click_url', this._storage),
+		click_id = utils.readKeyValue('click_id', this._storage);
```

Comment:
Well, let's think about this. If we're calling open with a match_id, it should always return a click_url, correct? In that case, we don't need to bother saving click_id.
The only other place click_id is saved is in sendSMS, but we could instead build a URL out of it and just save it as click_url instead. Then we could entirely do away with saving click_id.